File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        start = 0
        end = n - 1
        for i in range(n // 2):
            for j in range(start, end):
                matrix[i][j], matrix[j][end] = matrix[j][end], matrix[i][j]
                matrix[i][j], matrix[end][end - j + start] = matrix[end][end - j + start], matrix[i][j]
                matrix[i][j], matrix[end - j + start][i] = matrix[end - j + start][i], matrix[i][j]
            start += 1
            end -= 1
        

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        int start = 0;
        int end = n - 1;
        for(int i = 0; i < n / 2; i++)
        {
            for(int j = start; j < end; j++)
            {
                swap(matrix[i][j], matrix[j][end]);
                swap(matrix[i][j], matrix[end][end - j + start]);
                swap(matrix[i][j], matrix[end - j + start][i]);
            }
            start++;
            end--;
        }
    }
};